### ISSUE TITLE
MissingStubError when mocking DateTime: 'hashCode' / '==' / 'toString'

### ISSUE THREAD

**Issue by boldt** [dart-lang/mockito#420]
We're trying to migrate our mockito-tests to null-safety. We followed the [null-safety guide](https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md).
Now we see the following issues, when we are "working" with the stubs (like `expect` or `indexOf` in an array):

**1) MissingStubError: 'hashCode'**

```
MissingStubError: 'hashCode'
No stub was found which matches the arguments of this method call:
hashCode
```

**2) MissingStubError: '=='**

```
MissingStubError: '=='
No stub was found which matches the arguments of this method call:
==(MockDeviceModel)
```

**3) MissingStubError: 'toString'**

```
MissingStubError: 'toString'
No stub was found which matches the arguments of this method call:
toString()
```

**Our use case**

Our codes works on lists. Thereore, our tests use a list of mocks and thus, tried to compare them, which fails finally.

**Our setup**

* dart: 2.13.0
* flutter: 2.2.0
* mockito: 5.0.8

**Issue**

Mocks do not handle `hashCode` / `==` / `'toString'`

**Comment by boldt** [comment]
**MWE for MissingStubError: 'toString'**

```
@GenerateMocks([
  DateTime,
])
void main() {
  test('test', () {
    var mockDateTime = MockDateTime();
    expect(mockDateTime, mockDateTime);
  });
}
```

Error:

```
MissingStubError: 'toString'
No stub was found which matches the arguments of this method call:
toString()
```

**Comment by boldt** [comment]
# MWE for MissingStubError: 'toString'

```
  test("MissingStubError: 'hashCode'", () {
    var mockDateTime0 = MockDateTime();
    var mockDateTime1 = MockDateTime();
    var arr0 = [mockDateTime0, mockDateTime1];
    var arr1 = [mockDateTime0, mockDateTime1];
    expect(arr0, arr1);
  });
```

**Issue**

```
MissingStubError: 'hashCode'
No stub was found which matches the arguments of this method call:
hashCode
```

**Comment by srawlins** [comment]
I _think_ this might happen for any class which overrides toString, hashCode, or operator==, but it would be surprising that we haven't seen this error yet.

In any case, we _do_ have bugs for classes which override these methods and add optional parameters, so we should fix the two in one go.

**Comment by srawlins** [comment]
This should be fixed in mockito 5.0.10.

**Comment by boldt** [comment]
I can confirm, that my borken tests succeed now.
